src/parser/jobs/sam/modules/Kaiten.tsx

```diff
@@ -52,37 +52,40 @@ export class Kaiten extends Analyser {
 		super.initialise()
 
 		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		// If before 6.1, we need them hooks for kaiten.
+		if (this.parser.patch.before('6.1')) {
```

Comment:
I'd include the player filter in this, but consider just returning early if it's 6.1 or later instead.

Reply:
is there a difference besides style for returning early vs checking before moving on?

Reply:
Return early isn't really a pattern for this use case, I just felt it was better than wrapping a whole module in a giant condition, especially when the onComplete function will never be called if the hooks aren't set. Prob the cleanest way is to roll all of the init function into the condition, and then leave onComplete alone. Might be worth asking ack tho.